orchestrator/orchestrator_test.go

```diff
@@ -20,45 +17,121 @@ type TestSuite struct {
 	orchestrator *Orchestrator
 }
 
-func createTestSuite(t *testing.T) *TestSuite {
+type TestCase struct {
+	name    string
+	l1Host  string
+	l2Host  string
+	wantErr bool
+}
+
+func createTestSuite(t *testing.T, tc *TestCase) *TestSuite {
 	networkConfig := config.GetDefaultNetworkConfig(uint64(time.Now().Unix()), "")
 	testlog := testlog.Logger(t, log.LevelInfo)
 
+	l1Host := defaultHost
+	l2Host := defaultHost
+	if tc != nil {
+		l1Host = tc.l1Host
+		l2Host = tc.l2Host
+	}
+
+	networkConfig.L1Config.Host = l1Host
+	networkConfig.L1Config.Port = 9471
+	for i := range networkConfig.L2Configs {
+		networkConfig.L2Configs[i].Host = l2Host
+		networkConfig.L2StartingPort = 9473
+	}
```

Comment:
can we leave the starting ports to remain default 0? The test run in parallel so it will clash on port usage.